doc/user_guide/_docs/A20-logic-arrays.md

```diff
@@ -46,3 +46,36 @@ Array ports in generated SystemVerilog will match dimensions (including unpacked
 ## Elements of arrays
 
 To iterate through or access elements of a `LogicArray` (or bits of a simple `Logic`), use [`elements`](https://intel.github.io/rohd/rohd/Logic/elements.html).  Using the normal `[n]` accessors will return the `n`th bit regardless for `LogicArray` and `Logic` to maintain API consistency.
+
+## Index-based Selection in an Array
+
+The [`selectIndex`](https://intel.github.io/rohd/rohd/IndexedLogic/selectIndex.html) and [`selectFrom`](https://intel.github.io/rohd/rohd/Logic/selectFrom.html) methods are used to select a value from a `LogicArray` or from a list of `Logic` elements based on an index. These methods are useful for creating dynamic selection logic in hardware design. They can be used in 2 ways as shown below. 
+
+### 1. Using a `LogicArray` type
+
+```dart
+final arrayA = LogicArray([4], 8, name: 'arrayA'); // A 1D array with four 8-bit element
+final id = Logic(name: 'id', width: 3);
+
+selectIndexValueArrayA <= arrayA.elements.selectIndex(id, defaultValue: defaultValue);
+selectFromValueArrayA <= id.selectFrom(arrayA.elements, defaultValue: defaultValue);
+```
+An example code is given to demostrate a usage of selectIndex and selectFrom for logic arrays.
+Please see code here: [example/logic_array.dart](https://github.com/intel/rohd/blob/main/example/logic_array.dart)
```

Comment:
Sorry for the churn, but this may actually hit a link check failure since the example doesn't exist yet.  Perhaps better is to just mention it's in the ROHD `example` directory without a link?  Or just link to the `example` directory in general?